CAMEL-10509: ManagedCamelContextMBean - additional namespaces are removed

jamesnetherton opened the pull request:
https://issues.apache.org/jira/browse/CAMEL-10509

apupier commented on camel-core/src/main/java/org/apache/camel/model/ModelHelper.java:
```diff
@@ -110,13 +163,154 @@ public static String dumpModelAsXml(CamelContext context, NamedNode definition)
+        }
+
+        // Restore namespaces to anything that's NamespaceAware
+        if (result instanceof RoutesDefinition) {
```
duplicate block of code with what is l.135-145

apupier commented on camel-core/src/main/java/org/apache/camel/model/ModelHelper.java:
```diff
@@ -110,13 +163,154 @@ public static String dumpModelAsXml(CamelContext context, NamedNode definition)
+
+            NamespaceAware na = null;
+            Expression exp = ed.getExpressionValue();
+            if (exp != null && exp instanceof NamespaceAware) {
```
you can check only exp instanceof NamespaceAware because "null instanceof NamespaceAware" is returning false

apupier commented on camel-core/src/main/java/org/apache/camel/model/ModelHelper.java:
```diff
@@ -110,13 +163,154 @@ public static String dumpModelAsXml(CamelContext context, NamedNode definition)
+            ExpressionDefinition ed = en.getExpression();
+
+            // java-dsl sets directly expression so try this first
+            NamespaceAware na = null;
```
duplicated code with l.242-248

davsclaus commented on camel-core/src/main/java/org/apache/camel/model/ModelHelper.java:
```diff
@@ -53,12 +70,38 @@ public static String dumpModelAsXml(CamelContext context, NamedNode definition)
+            documentElement.setAttribute("xmlns:" + nsPrefix, namespaces.get(nsPrefix));
+        }
+
+        Exchange exchange = ExchangeBuilder.anExchange(context).build();
```
Very clever. But I would like to see a way to convert this without having to create a dummy exchange. Maybe you can add some methods on the xml converter to accept those indent options or something and call the converter directly.

Creating an exchange is really only intended for routing messages.

davsclaus commented:
Great work James, only a comment about that convert to xml using Exchange trick you do.

jamesnetherton commented:
Pushed up some updates...

* Removed the Exchange property hack and now invoke the type converter directly
* Cleaned up some of the duplication in ModelHelper. I didn't fix up everything. Probably something that could be done in future as part of a larger refactoring.

davsclaus commented:
Thanks almost there, some unit tests in camel-spring now fails. Can you fix this also

```
Failed tests:
org.apache.camel.spring.DumpModelAsXmlPlaceholdersTest.testDumpModelAsXml(org.apache.camel.spring.DumpModelAsXmlPlaceholdersTest)
  Run 1: DumpModelAsXmlPlaceholdersTest>TestSupport.runBare:58->testDumpModelAsXml:39
  Run 2: DumpModelAsXmlPlaceholdersTest>TestSupport.runBare:58->testDumpModelAsXml:39
  Run 3: DumpModelAsXmlPlaceholdersTest>TestSupport.runBare:58->testDumpModelAsXml:39
```

jamesnetherton commented:
Sorry about that - should be fixed now.

davsclaus commented:
Thanks for the PR it has been merged. Do you mind closing this?
